File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int maxl=0;
        for(int i=0;i<s.size();i++){
            vector<int> hash(256,0);

            for(int j=i;j<s.size();j++){
                if(hash[s[j]]==1){
                    break;
                }
                hash[s[j]]=1;
                int len=j-i+1;
                maxl=max(len,maxl);
            }
            
        }
        return maxl;
        
    }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int maxi=0;
        HashMap<Character,Integer> mpp = new HashMap<>();
        int i=0;
        for(int j=0;j<s.length();j++){
            if(!mpp.containsKey(s.charAt(j))|| mpp.get(s.charAt(j))<i){
                mpp.put(s.charAt(j),j);
                maxi=Math.max(maxi,j-i+1);
            }else{
                i=mpp.get(s.charAt(j))+1;
                mpp.put(s.charAt(j),j);
            }
        }
        return maxi;
    }
}